runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -209,6 +210,13 @@ public void addTransaction(AbstractTransactionalContext tc) {
     @Override
     @SuppressWarnings("unchecked")
     public long commitTransaction() throws TransactionAbortedException {
+
+        return commitTXHelper(() -> getReadSet());
+    }
+
+    public long commitTXHelper(Supplier<Map<UUID, Set<Integer>>>
```

Comment:
This needs to be named something better than commitTXHelper. Maybe doOptimisticCommit?